homeassistant/components/fibaro/sensor.py

```diff
@@ -114,3 +132,52 @@ def update(self):
         """Update the state."""
         with suppress(KeyError, ValueError):
             self.current_value = float(self.fibaro_device.properties.value)
+
+
+class FibaroEnergySensor(FibaroDevice, SensorEntity):
+    """Representation of a Fibaro Energy Sensor."""
+
+    def __init__(self, fibaro_device):
+        """Initialize the sensor."""
+        self.current_value = None
+        self.last_changed_time = None
+        super().__init__(fibaro_device)
+        self.entity_id = f"{DOMAIN}.{self.ha_id}_energy"
+        self._name = f"{fibaro_device.friendly_name} Energy"
+
+        self._unit = ENERGY_KILO_WATT_HOUR
+        self._icon = None
+        self._device_class = SensorDeviceClass.ENERGY
+        self._state_class = SensorStateClass.TOTAL_INCREASING
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        return self.current_value
+
+    @property
+    def native_unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return self._unit
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return self._device_class
+
+    @property
+    def state_class(self):
+        """Return the state class of the sensor."""
+        return self._state_class
+
+    def update(self):
+        """Update the state."""
+        with suppress(KeyError, ValueError):
+            self.current_value = convert(
+                self.fibaro_device.properties.energy, float, 0.0
```

Comment:
I don't think we should default this to 0.0... its means you all of sudden haven't used energy anymore when an issue occurs... which isn't true.

Reply:
Removed default 0 (which means None is used). Just copied this bad code from another place.